Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs

```diff
@@ -315,6 +335,9 @@ public ActionResult Retest(int id)
                     return this.RedirectToAction("Index", "Contests", new { area = "Administration" });
                 }
 
+                this.Data.ParticipantScores.DeleteParticipantScore(submission);
+                this.Data.SaveChanges();
```

Comment:
An action's DB modification should be executed as a single transaction, to ensure no invalid state in the DB can be reached